plugins/inputs/smart/smart.go

```diff
@@ -282,16 +304,96 @@ func gatherDisk(acc telegraf.Accumulator, usesudo, attributes bool, smartctl, no
 			// save the raw value to a field.
 			if field, ok := deviceFieldIds[attr[1]]; ok {
 				if val, err := parseRawValue(attr[8]); err == nil {
-					device_fields[field] = val
+					deviceFields[field] = val
 				}
 			}
+		} else {
+			var tempC int64
+			if collectAttributes {
+				if startStop := sasStartStopAttr.FindStringSubmatch(line); len(startStop) > 1 {
```

Comment:
There are equivalent smart attribute fields for each of these though. They are just collected differently for these different device types.

Reply:
Attributes would have the id, name, flags, raw value and threshold in a 8bit scale with 100 being the default, etc.

I guess this is okay though, I am a little uncomfortable with the information we are adding, but it will be better than having too many vendor specific fields on the device.